docs/UNICODE_NOTES.md

```diff
@@ -0,0 +1,170 @@
+| Egyptian Hieroglyphs | 𓀀 𓆉 𓍄 | 2 |
+| Musical Symbols | 𝄞 𝄢 | 2 |
+| Mathematical Alphanumeric | 𝔸 𝕏 | 2 |
+| Historic Scripts | Various | 2 |
+
+### Characters in BMP (No Difference)
+
+| Category | Examples | UTF-16 Units per Char |
+|----------|----------|----------------------|
+| ASCII/Latin | A-Z, a-z, 0-9 | 1 |
+| Latin Extended | á, ñ, ü, ø | 1 |
+| Chinese | 中文字 | 1 |
+| Japanese (Hiragana/Katakana/Kanji) | 日本語 | 1 |
+| Korean (Hangul) | 한글 | 1 |
+| Arabic | العربية | 1 |
+| Hebrew | עברית | 1 |
+| Cyrillic | русский | 1 |
+| Greek | ελληνικά | 1 |
+| Thai | ไทย | 1 |
+
+## Our Solution: WASM-Based Validation
+
+All validation in `pubky-app-specs` happens **inside the WASM module** (Rust), not in JavaScript.
+
+### Architecture
+
+```
+┌─────────────────────────────────────────────────────────┐
+│                    JavaScript Client                     │
+│                                                         │
+│   const user = PubkyAppUser.fromJson({                  │
+│       name: "Alice🔥",                                  │
+│       bio: "Hello 𓀀"                                   │
+│   });                                                   │
+└─────────────────────┬───────────────────────────────────┘
+                      │
+                      ▼
+┌─────────────────────────────────────────────────────────┐
+│                    WASM Module (Rust)                    │
+│                                                         │
+│   1. Deserialize JSON                                   │
+│   2. Sanitize (trim whitespace, normalize)              │
+│   3. Validate (using .chars().count())  ◄── Single      │
+│   4. Return Result                           Source     │
+│                                              of Truth   │
+└─────────────────────────────────────────────────────────┘
+```
+
+### Why This Works
+
+1. **Single Source of Truth**: All validation uses Rust's `.chars().count()`, which counts Unicode code points.
+2. **No JS Validation**: JavaScript never validates string lengths directly—it delegates to WASM.
+3. **Consistent Behavior**: Whether the user types emoji, Chinese, or cuneiform, the validation is consistent.
+
+### Example: Username Validation
+
+```rust
+// In Rust (WASM)
+const MAX_USERNAME_LENGTH: usize = 50;
+
+fn validate(&self, _id: Option<&str>) -> Result<(), String> {
+    let name_length = self.name.chars().count();  // Unicode code points
+    if name_length > MAX_USERNAME_LENGTH {
+        return Err("Validation Error: Invalid name length".into());
+    }
+    Ok(())
+}
+```
+
+| Input | `.chars().count()` | Valid? (max 50) |
+|-------|-------------------|-----------------|
+| `"Alice"` | 5 | ✅ |
+| `"🔥".repeat(50)` | 50 | ✅ |
+| `"🔥".repeat(51)` | 51 | ❌ |
+| `"𓀀".repeat(50)` | 50 | ✅ |
+
+## Important: Don't Validate in JavaScript
+
+If you need client-side validation (for UX feedback), you **must** match Rust's behavior or trust in pubk-app-specs WASM module.
```

Comment:
> or trust in pubk-app-specs WASM module

First, seems like a weird ask. Maybe it should be rephrased, for example "or rely on the existing pubky-app-specs validation in the WASM module".

Second, the snippet below doesn't show how the caller would do that. It shows good-enough JS approximations of the same validation logic, but
- A) 50 is hardcoded, it could change, so it's not a very good idea to recommend this
- B) it's not clear how JS could call the WASM-based validation, for example a call to `User.validate()`

So IMO the correct snippets below have to be replaced with an example of how to call the correct validation.